### ISSUE TITLE
@babel/node not recognizing ignores from babel.config.js when running babel-node

### ISSUE THREAD

**Issue by jcreamer898** [babel/babel#11892]
## Bug Report

**Current behavior**
It's possible it's a "feature", but seems like a bug.

So, I have a babel config that has an ignore function in it. Mostly because where I work we have a couple of internal node modules we publish that have some `dist` folders with `import` `export` statements.

The `babel.config.js` looks more or less like this...

```js
module.exports = (api) => {
    api.cache(true);

    return {
        presets: ['@babel/env'],
        ignore: [function(filepath) {
            if (filepath.includes('my-npm-pkg')) {
                return false;
            }

            return filepath.includes('node_modules');
        }]
    };
};
```

When I run the following...

```shell
babel-node ./index.js
```

I get the following error...

![](https://d.pr/i/41rux3+)

However, if I run it with an empty ignore, like this...

```shell
babel-node --ignore ' ' ./index.js
```

Everything works...

![](https://d.pr/i/41rux3+)

**Expected behavior**
I'd expect that the ignore should be taken from the `babel.config.js` even if I don't have an `--ignore` in the command line args.

**Environment**
I'm running on node 12.17.0.

```
"@babel/core": "^7.10.5",
"@babel/node": "^7.10.5",
"@babel/preset-env": "^7.10.4"
```

I also created an example repository here... 

https://github.com/jcreamer898/babel-node-ignore

**Possible Solution**
I'm not entirely sure, but maybe it has something to do with https://github.com/babel/babel/blob/main/packages/babel-register/src/node.js#L140? 

Does there need to be a check for the babel config's ignore here? It didn't _seem_ like that to me because when I console logged the options getting passed here, it didn't even seem to have the ignore from my `babel.config.js`.

https://github.com/babel/babel/blob/main/packages/babel-node/src/_babel-node.js#L94

![](https://d.pr/i/o9m7Iq+)

So, my suspicion is that the `--ignore ' '` is just nuking the ignore option totally, and `babel.config.js` is firing on a file by file basis later on as they're processing individual files?

**Comment by jcreamer898** [comment]
For what it's also worth I finally got an ignore regex to work on the command line as well, just took me a while to get it right...

```shell
babel-node --ignore '\/node_modules\/(?!my-npm-pkg)\/' index.js
```

**Comment by jcreamer898** [comment]
Oops, sorry, sec.

**Comment by jcreamer898** [comment]
Fixed.

**Comment by sz-coder** [comment]
Not sure if I understand correctly but shouldn't `babel-node --config-file ./babel.config.js ./index.js` yield the same result as `--ignore '\/node_modules\/(?!my-npm-pkg)\/'` ?

**Comment by sz-coder** [comment]
I played around with your `babel.config.js` a bit... seems like it doesn't matter whether the function returns `true` or `false`, both yield the same error. 🤔

**Comment by jcreamer898** [comment]
Right, exactly, it's like it doesn't work at all when the config file is being used. I think the "default" --ignore is taking over whatever is in the babel config.

**Comment by Edmond-XavierCollot** [comment]
I have the same issue, the `only` and `ignore` options of my `babel.config.json` are ignored when running `babel-node`.
I have to specify the options directly in the cli 
```babel-node --only src,my-library src/index.js```
